sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -1838,6 +1838,12 @@ object SQLConf {
     .booleanConf
     .createWithDefault(false)
 
+  val LEGACY_CTE_SUBSTITUTION_ENABLED = buildConf("spark.sql.legacy.cte.substitution.enabled")
```

Comment:
We still do `substituteCTE`. Can we have a better name?

Reply:
How about `spark.sql.legacy.outerCTEHasPriority`? Please let me know if you have a better name.

Reply:
`spark.sql.legacy.ctePrecedence.enabled` or `spark.sql.legacy.cteResolutionPrecedence.enabled`?

Reply:
Both sound fine to me. @dongjoon-hyun?